kuma/settings/common.py

```diff
@@ -406,32 +406,33 @@ def enable_candidate_languages():
 )
 
 MIDDLEWARE_CLASSES = (
-    # must come before LocaleURLMiddleware
     'redirect_urls.middleware.RedirectsMiddleware',
-    # LocaleURLMiddleware must be before any middleware that uses
-    # kuma.core.urlresolvers.reverse() to add locale prefixes to URLs:
     'kuma.core.middleware.SetRemoteAddrFromForwardedFor',
     ('kuma.core.middleware.ForceAnonymousSessionMiddleware'
      if MAINTENANCE_MODE else
      'django.contrib.sessions.middleware.SessionMiddleware'),
+
+    # LocaleURLMiddleware requires access to request.user. These two must be
+    # loaded before the LocaleURLMiddleware
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'kuma.core.anonymous.AnonymousIdentityMiddleware',
```

Comment:
Rearranging middleware seems dangerous to me.  Not sure if our tests would fully expose issues.